### PR TITLE
give people downstream a prayer of identifying what messages arbitrar…

### PR DESCRIPTION
**Pull request by osh**
…y parsing errors come from by allowing them to also obtain the raw xml from the message in question

### CODE REVIEW

**Comment by osh** [comment]
since you pass the raw http response object, there is not another way for people downstream to get the raw xml for logging/error diagnosis purposes ..... this needs to be accessible somehow!

**Comment by gtback** [comment]
Hi, @osh . Thanks for the PR!

I'm pretty sure this will break all existing uses of the `get_message_from_*` functions, including the scripts included with this library (in addition to any external code). At a minimum, the scripts need to be updated, but ideally, my preferred approach would be to introduce this functionality in new methods (which steal the current functionality of the existing functions), and then make the current functions into wrappers around the new functions that return the second item in each tuple, to match the current interface.

Also, do you have an example of how you would use the new data returned from these functions? I know our documentation isn't great, but I'd like to improve it if possible. 😀 

Thanks again!

**Comment by gtback** [comment]
@osh, have you had a chance to make any of the requested changes? We are probably going to release a new version of libtaxii in the next couple weeks, so wanted to get your changes into that release if possible.

**Comment by osh** [comment]
haven't yet --- 
I'm not a huge fan of creating parallel alternate function calls and exploding the API
perhaps passing an arg   (return_xml=False)  as default, which can be set to true to return a tuple instead of just the object would be a reasonable approach to backwards compatibility?

**Comment by osh** [comment]
this should now be API backwards compatible and ready for merge

**Comment by gtback** [comment]
Is there a good reason to return just the response message and not the entire response? For logging/error checking purposes, I imagine the latter would be more useful.

**Comment by gtback** [comment]
Also, and this is an *extremely* minor thing, changes should be PEP-8 compatible (spaces after commas). We don't have good automated checking of that at the moment, and there's lots of code in the libtaxii that isn't PEP-8 compatible, but I'd like any future changes/enhancements to be. Thanks!

File: libtaxii/__init__.py [hunk 1]
```diff
@@ -52,8 +52,13 @@ def get_message_from_http_response(http_response, in_response_to):
     else:
         raise ValueError('Unsupported response type: %s.' % http_response.__class__.__name__)
 
+def return_check(return_message,message,object):
```
Comment (MarkDavidson) [inline libtaxii/__init__.py]: This has a code smell to it - Can the function be refactored to always return the same type?

**Comment by MarkDavidson** [comment]
IMO the underlying difficulty is that HTTPResponses generally have a `.read()` function that is consumable - you can only use it once.

I agree with the goal of the PR - but perhaps there is a different way to go about making the changes.

A few comments:

* I made an earlier comment that I deleted because it was incorrect. Sorry about that.

* https://github.com/TAXIIProject/libtaxii/blob/master/libtaxii/messages_11.py#L79 - This is where any parse errors would occur. This seems like the right place to have potential debug/logging, as it is where parse errors would be raised.
  * I could see adding a `debug=False` kwarg to the function, and if `debug=True` it prints/logs the xml

* Errors in well-formed but incorrect XML would have to be handled differently

Thank you.
-Mark

File: libtaxii/__init__.py (hunk 1 above)
Comment (gtback) [inline libtaxii/__init__.py]: That is what I suggested [here](https://github.com/TAXIIProject/libtaxii/pull/213#issuecomment-256759292), and I maintain that it is a better approach than the `return_check` function.

**Comment by gtback** [comment]
I understand the goal of this PR, but I won't merge it in in the current state. Although the current approach is backwards-compatible, it adds duplicate code beyond what I'm comfortable with. 

I'd be happy to consider an alternate implementation along the lines of what I outlined. If others in addition to @osh are interested in this functionality, please comment or 👍 this PR.

**Pull request closed without merging**